reference-architecture/day2ops/scripts/project_export.sh

```diff
@@ -31,48 +31,60 @@ usage(){
   warnuser
 }
 
-ns(){
-  echo "Exporting namespace to ${PROJECT}/ns.json"
-  oc get --export -o=json ns/${PROJECT} | jq '
-    del(.status,
-      .metadata.uid,
-      .metadata.selfLink,
-      .metadata.resourceVersion,
-      .metadata.creationTimestamp,
-      .metadata.generation
-      )' > ${PROJECT}/ns.json
+exportlist(){
+  if [ "$#" -lt "3" ]; then
+    echo "Invalid parameters"
+    return
+  fi
+
+  KIND=$1
+  BASENAME=$2
+  DELETEPARAM=$3
+
+  echo "Exporting '${KIND}' resources to ${PROJECT}/${BASENAME}.json"
+
+  BUFFER=`oc get ${KIND} --export -o json -n ${PROJECT} || true`
+
+  # return if resource type unknown or access denied
+  if [ -z "${BUFFER}" ]; then
+    echo "Skipped: no data"
+    return
+  fi
+
+  # return if list empty
+  if [ `echo ${BUFFER} | jq '.items | length > 0'` == "false" ]; then
+    echo "Skipped: list empty"
+    return
+  fi
+
+  echo ${BUFFER} | jq ${DELETEPARAM} > ${PROJECT}/${BASENAME}.json
 }
 
+ns(){
+  exportlist \
+    ns \
+    ns \
+    'del(.items[].status,.items[].metadata.uid,.items[].metadata.selfLink,.items[].metadata.resourceVersion,.items[].metadata.creationTimestamp,.items[].metadata.generation)'
+}
 rolebindings(){
-  echo "Exporting rolebindings to ${PROJECT}/rolebindings.json"
-  oc get --export -o=json rolebindings -n ${PROJECT} | jq '.items[] |
-  del(.metadata.uid,
-      .metadata.selfLink,
-      .metadata.resourceVersion,
-      .metadata.creationTimestamp
-      )' > ${PROJECT}/rolebindings.json
+  exportlist \
+    rolebindings \
+    rolebindings \
+    'del(.items[].metadata.uid,.items[].metadata.selfLink,.items[].metadata.resourceVersion,.items[].metadata.creationTimestamp)'
 }
 
 serviceaccounts(){
-  echo "Exporting serviceaccounts to ${PROJECT}/serviceaccounts.json"
-  oc get --export -o=json serviceaccounts -n ${PROJECT} | jq '.items[] |
-    del(.metadata.uid,
-        .metadata.selfLink,
-        .metadata.resourceVersion,
-        .metadata.creationTimestamp
-        )' > ${PROJECT}/serviceaccounts.json
+  exportlist \
+    serviceaccounts \
+    serviceaccounts \
+    'del(.items[].metadata.uid,.items[].metadata.selfLink,.items[].metadata.resourceVersion,.items[].metadata.creationTimestamp)'
```

Comment:
And this (and any other list to avoid commenting too much)

Reply:
Fixed in 02cfd39.